products/compute/api.yaml

```diff
@@ -7476,6 +7486,147 @@ objects:
+        name: name
+        description: The name of the packet mirroring rule
+        required: true
+      - !ruby/object:Api::Type::String
+        name: description 
+        description: A human-readable description of the rule.
+        input: true
+      - !ruby/object:Api::Type::String
+        name: region
+        description: The region where this rule is active.
+        required: true
+      - !ruby/object:Api::Type::NestedObject
+        name: network
+        description: |
+            Specifies the mirrored VPC network. Only packets in this network
+            will be mirrored. All mirrored VMs should have a NIC in the given
+            network. All mirrored subnetworks should belong to the given network.
+        properties:
+          - !ruby/object:Api::Type::String
+            name: url
+            description: The full self_link URL of the network where this rule is active.
+            required: true
+        required: true
+      - !ruby/object:Api::Type::Integer
+        name: priority
+        description: |
+            Since only one rule can be active at a time, priority is
+            used to break ties in the case of two rules that apply to
+            the same instances.
+        required: true
+      - !ruby/object:Api::Type::NestedObject
+        name: collectorIlb
+        description: |
+            The Forwarding Rule resource (of type loadBalancingScheme=INTERNAL)
+            that will be used as collector for mirrored traffic. The
+            specified forwarding rule must have isMirroringCollector
+            set to true.
+        required: true
+        properties:
+          - !ruby/object:Api::Type::String
+            name: url
+            required: true
+            description: The URL of the forwarding rule.
+      - !ruby/object:Api::Type::NestedObject
+        name: filter 
+        description: |
+            A filter for mirrored traffic.  If unset, all traffic is mirrored.
+        properties:
+          - !ruby/object:Api::Type::Array
+            name: ipProtocols
+            api_name: 'IPProtocols'
+            description: |
+                Protocols that apply as a filter on mirrored traffic.
+            item_type: !ruby/object:Api::Type::Enum
+              name: ipProtocols
+              description: Possible IP protocols to filter on.
+              values:
+                - tcp
+                - udp
+                - icmp
+          - !ruby/object:Api::Type::Array
+            name: cidrRanges
+            description: |
+                IP CIDR ranges that apply as a filter on the source (ingress) or
+                destination (egress) IP in the IP header. Only IPv4 is supported.
+            item_type: Api::Type::String
+      - !ruby/object:Api::Type::NestedObject
+        name: mirroredResources
+        required: true
+        description: |
+            A means of specifying which resources to mirror.
+        at_least_one_of:
```

Comment:
I'm not sure the syntax is correct on this: https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-3157-old..auto-pr-3157#diff-3ab2ba91e18e29d378d69f103ff962d0R108